ksrates/wgd_paralogs.py

```diff
@@ -16,35 +16,43 @@ def wgd_paralogs(config_file, n_threads):
     init_logging(f"Paralog wgd analysis for species [{species}]", config.get_logging_level())
 
     latin_names = config.get_latin_names()
-    fasta_names_dict = config.get_fasta_dict()
-    species_fasta_file = config.get_fasta_name(fasta_names_dict, species)
-    fcCheck.check_inputfile(species_fasta_file, "FASTA file")
     max_gene_family_size = config.get_max_gene_family_size()
-
     paranome = config.get_paranome()
     colinearity = config.get_colinearity()
 
-    if colinearity:  # if colinearity analysis is required, load extra parameters
-        gff_names_dict = config.get_gff_dict()
-        species_gff_file = config.get_gff_name(gff_names_dict, species)
-        fcCheck.check_inputfile(species_gff_file, "GFF file")
+    logging.info(f"Checking if sequence data files exist and if sequence IDs are compatible with wgd pipeline...")
+    # Will exit if FASTA or GFF files are missing or empty or if GFF feature/attribute are missing
+    trigger_exit = False
+
+    if colinearity:  # if colinearity analysis is required, load related parameters
+        gff = config.get_gff(species)
+        if fcCheck.check_file_nonexistent_or_empty(gff, "GFF file"):
+            trigger_exit = True
 
         gff_feature = config.get_feature()
         gff_gene_attribute = config.get_attribute()
         if gff_feature == "":
-            logging.error("No GFF attribute provided in configuration file.")
+            logging.error("No GFF attribute provided in configuration file. Will exit.")
+            trigger_exit = True
         if gff_gene_attribute == "":
-            logging.error("No GFF feature provided in configuration file.")
-        if gff_feature == "" or gff_gene_attribute == "":
-            logging.error("Will exit the colinearity analysis.")
-            sys.exit(1)
-    # Checking if IDs in FASTA (and in GFF if applicable) are compatible with wgd pipeline (paml)
-    logging.info("")
-    if colinearity:
-        fcCheck.check_IDs(species_fasta_file, latin_names[species], species_gff_file)
-    else:
-        fcCheck.check_IDs(species_fasta_file, latin_names[species])
+            logging.error("No GFF feature provided in configuration file. Will exit.")
+            trigger_exit = True
+
+    # Checking if FASTA file exists and if sequence IDs are compatible with wgd pipeline (paml)
+    fasta_names_dict = config.get_fasta_dict()
+    species_fasta_file = config.get_fasta_name(fasta_names_dict, species)
+    if fcCheck.check_file_nonexistent_or_empty(species_fasta_file, "FASTA file"):  # if missing/empty
+        trigger_exit = True
```

Comment:
I would have probably left doing the FASTA file check first before the GFF file check.